src/sysupdate/sysupdate-transfer.c

```diff
@@ -165,6 +168,44 @@ static int config_parse_min_version(
         return free_and_replace(*version, resolved);
 }
 
+static int config_parse_url_specifiers(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        char ***s = ASSERT_PTR(data);
+        _cleanup_free_ char *resolved = NULL;
+        int r;
+
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                *s = mfree(*s);
```

Comment:
If we keep this as a strv, then you must `strv_free` here otherwise we're leaking memory